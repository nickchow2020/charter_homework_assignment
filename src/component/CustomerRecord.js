const CustomerRecord = ({ record }) => {
  return (
    <>
      <tr>
        <td>{record.buyerId}</td>
        <td>{record.transactionNumber}</td>
        <td>{record.name}</td>
        <td>{record.purchaseItem}</td>
        <td>{record.price}</td>
        <td>{record.rewardPoints}</td>
        <td>{record.transactionDate}</td>
      </tr>
    </>
  );
};

export default CustomerRecord;
