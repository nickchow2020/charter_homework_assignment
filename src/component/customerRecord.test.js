import { render, screen } from '@testing-library/react';
import CustomerRecord from './CustomerRecord';

describe('<CustomerRecord />', () => {
  const MockData = {
    buyerId: 'test1',
    transactionNumber: 'test2',
    itemI: 'test3',
    itemName: 'test4',
    price: 100,
    rewardPoints: 50,
    transactionDate: 'May 11 2022',
  };

  it('should match snapshot', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <CustomerRecord record={MockData} />
        </tbody>
      </table>
    );
    expect(asFragment).toMatchSnapshot();
  });

  it('should contain test1', () => {
    render(
      <table>
        <tbody>
          <CustomerRecord record={MockData} />
        </tbody>
      </table>
    );
    const targetText = screen.getByText('test1');

    expect(targetText).toBeInTheDocument();
  });

  it('should contain May 11 2022', () => {
    render(
      <table>
        <tbody>
          <CustomerRecord record={MockData} />
        </tbody>
      </table>
    );
    const targetText = screen.getByText('May 11 2022');

    expect(targetText).toBeInTheDocument();
  });
});
