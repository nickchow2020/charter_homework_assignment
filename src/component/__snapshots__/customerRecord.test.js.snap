// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomerRecord /> should match snapshot 1`] = `[Function]`;
